### ISSUE TITLE
Fix evaluate function for multi-class mode

### ISSUE THREAD

**Issue by valhassan** [NRCan/geo-deep-learning#395]
This issue is similar to #394 we would like to apply a **softmax** followed by **argmax** on output logits before calculating metrics on corresponding labels. This assumes the model does not use a softmax layer before outputs are returned.
